packages/squiggle-lang/src/rescript/Distributions/GenericDist.res

```diff
@@ -159,6 +119,89 @@ let toPointSet = (
   }
 }
 
+module Score = {
+  type scoreDistOrScalar = DistributionTypes.DistributionOperation.scoreDistOrScalar
+
+  type pointSet_ScoreDistOrScalar = D(PointSetTypes.pointSetDist) | S(float)
+  let argsMake = (
+    ~esti: scoreDistOrScalar,
+    ~answ: scoreDistOrScalar,
+    ~prior: option<scoreDistOrScalar>,
+  ): result<PointSetDist_Scoring.scoreArgs, error> => {
+    let toPointSetFn = toPointSet(
+      ~xyPointLength=MagicNumbers.Environment.defaultXYPointLength,
+      ~sampleCount=MagicNumbers.Environment.defaultSampleCount,
+      ~xSelection=#ByWeight,
+    )
+    let prior': option<result<pointSet_ScoreDistOrScalar, error>> = switch prior {
+    | None => None
+    | Some(Score_Dist(d)) => toPointSetFn(d, ())->E.R.bind(x => x->D->Ok)->Some
+    | Some(Score_Scalar(s)) => s->S->Ok->Some
+    }
+    let twoDists = (esti': t, answ': t): result<
+      (PointSetTypes.pointSetDist, PointSetTypes.pointSetDist),
+      error,
+    > => E.R.merge(toPointSetFn(esti', ()), toPointSetFn(answ', ()))
+    switch (esti, answ, prior') {
+    | (Score_Dist(esti'), Score_Dist(answ'), None) =>
+      twoDists(esti', answ')->E.R.bind(((esti'', answ'')) =>
+        {estimate: esti'', answer: answ'', prior: None}
+        ->PointSetDist_Scoring.DistEstimateDistAnswer
+        ->Ok
+      )
+    | (Score_Dist(esti'), Score_Dist(answ'), Some(Ok(D(prior'')))) =>
+      twoDists(esti', answ')->E.R.bind(((esti'', answ'')) =>
+        {estimate: esti'', answer: answ'', prior: Some(prior'')}
+        ->PointSetDist_Scoring.DistEstimateDistAnswer
+        ->Ok
```

Comment:
I think this is using .bind() and adding Ok()? I imagine it could just use .fmap() instead?